src/parser/shared/normalizers/Channeling.ts

```diff
@@ -355,6 +355,53 @@ function buffChannelSpec(spellId: number): ChannelSpec {
   };
 }
 
+/**
+ * Helper to create a channel spec handler for the common case of a channeled spell that can be delimited by a buff.
+ * These cases involve a channeled spell that produces a Cast and ApplyBuff event (with the same guid)
+ * when it starts, and then a RemoveBuff event when it finishes. Some instead put a Debuff on the target, but the
+ * principle is the same.
+ *
+ * This handler works by handling cast events with the given guid, and then scanning forward for the
+ * matched removebuff (or removedebuff) event, as well as another cast or begincast event
+ * and then making the pair of beginchannel and endchannel events based on them.
+ *
+ * @param spellId the guid for the tracked Cast and RemoveBuff/RemoveDebuff events.
+ */
+function buffAndNextCastChannelSpec(spellId: number): ChannelSpec {
```

Comment:
i feel like in order to prevent bad worst-case behavior, this should have a max duration (if possible without breaking chained casts). but i can't come up with a scenario where it would break because we also reset on the next cast of the same ability. so the worst case scenario is one where you never cast it again and we never see a removebuff---in which case we just look through the remaining events once. so maybe it is fine as-is :thinking:

Reply:
It's only for the buff/debuff part where it checks for same spell, for casts it will reset on any cast. So it should be pretty safe.

Reply:
Is it going to reset on any cast? the handler is only called for the guids in the list. otherwise the default handler is called (which might actually end the channel. it does at least for `begincast`. not familiar enough to fully parse the `cast` branch)

Reply:
oh, yup, i see what you're saying now. :+1: i agree this seems safe